imports/api/blocks/server/methods.js

```diff
@@ -482,6 +481,79 @@ Meteor.methods({
 
                         }
 
+
+                        // check if there's any validator not in db 14400 blocks(~1 day)
+                        if (height % 14400 == 0){
+                            try {
+                                console.log('Checking all validators against db...')
+                                let dbValidators = {}
+                                Validators.find({}, {fields: {consensus_pubkey: 1, status: 1}}
+                                    ).forEach((v) => dbValidators[v.consensus_pubkey] = v.status)
+                                Object.keys(validatorSet).forEach((conPubKey) => {
+                                    let validator_data = validatorSet[conPubKey];
+                                    // Active validators should have been updated in previous steps
+                                    if (validator_data.status === 2)
+                                        return
+
+                                    if (dbValidators[conPubKey] == undefined) {
+                                        console.log(`validator with consensus_pubkey ${conPubKey} not in db`);
+
+                                        validator_data.pub_key = {
+                                            "type" : "tendermint/PubKeyEd25519",
+                                            "value": Meteor.call('bech32ToPubkey', conPubKey)
+                                        }
+                                        validator_data.address = getAddress(validator_data.pub_key);
+                                        validator_data.delegator_address = Meteor.call('getDelegator', validator_data.operator_address);
+
+                                        validator_data.accpub = Meteor.call('pubkeyToBech32', validator_data.pub_key, Meteor.settings.public.bech32PrefixAccPub);
+                                        validator_data.operator_pubkey = Meteor.call('pubkeyToBech32', validator_data.pub_key, Meteor.settings.public.bech32PrefixValPub);
+                                        console.log(JSON.stringify(validator_data))
+                                        bulkValidators.find({consensus_pubkey: conPubKey}).upsert().updateOne({$set:validator_data});
+                                    } else if (dbValidators[conPubKey] == 2) {
+                                        bulkValidators.find({consensus_pubkey: conPubKey}).upsert().updateOne({$set:validator_data});
+                                    }
+                                })
+                            } catch (e){
+                                console.log(e)
+                            }
+                        }
+
+                        // fetching keybase every 14400 blocks(~1 day)
+                        if (height % 14400 == 1){
```

Comment:
Possible to let validators to show their profile picture after they have updated their validator info? Validators expect to see it immediately. Store the profile url when a `create-validator` or `edit-validator` tx happen?

Reply:
ack